compiler/ccgstmts.nim

```diff
@@ -1147,5 +1147,11 @@ proc genAsgn(p: BProc, e: PNode, fastAsgn: bool) =
 
 proc genStmts(p: BProc, t: PNode) =
   var a: TLoc
+
+  let isPush = hintExtendedContext in p.config.notes
+  if isPush: pushInfoContext(p.config, t.info)
+  defer:
```

Comment:
using defer for correctness; see also https://forum.nim-lang.org/t/4022: "are nim exceptions "zero cost" (on error-free execution) ? (if so, `defer` should be used more)"

Reply:
Don't use ``defer`` nor try finally. This code cannot throw anyway.

Reply:
done, although I preferred the `defer` way (assuming `defer` gets fixed cf https://github.com/nim-lang/Nim/issues/8254), it's more robust in case future code changes make that code throw